pyquil/quilatom.py

```diff
@@ -35,24 +37,39 @@ def __init__(self, index):
     def out(self):
         return str(self.index)
 
+    def __str__(self):
+        return str(self.index)
+
     def __repr__(self):
         return "<Qubit {0}>".format(self.index)
 
+    def __hash__(self):
+        return hash(self.index)
+
+    def __eq__(self, other):
+        return isinstance(other, Qubit) and other.index == self.index
 
-class QubitPlaceholder(Qubit):
-    def __init__(self):
-        pass
 
-    @property
-    def index(self):
-        raise RuntimeError("Qubit has not been assigned an index")
+class QubitPlaceholder(QuilAtom):
+    def out(self):
+        raise RuntimeError("Qubit {} has not been assigned an index".format(self))
 
     def __str__(self):
         return repr(self)
 
     def __repr__(self):
         return "<QubitPlaceholder {}>".format(id(self))
 
+    def __hash__(self):
+        return hash(id(self))
+
+    def __eq__(self, other):
+        return isinstance(other, QubitPlaceholder) and id(other) == id(self)
+
+    @classmethod
+    def register(cls, n):
```

Comment:
If this is a public method that people are expected to use then it needs to be documented. Also, this doesn't seem like a very discoverable place for it (although I have no concrete suggestions of where else to put it).

Reply:
I want `QubitPlaceholder` to be a top level import eventually cc #302 which would help with discoverability.

Really, I wish we could "flip the script" and have `Qubit` be a placeholder / thing that needs to be addressed and rename `Qubit` to `PhysicalQubit` or something annoying (or drop it and just use python ints). Then this would be much nicer

```
q = Qubit.register(8)
[H(q[i]) for i in range(8)]
```

Reply:
Add it to the list for 2.0 😛